R/make-traceability-matrix.R

```diff
@@ -70,6 +70,12 @@ map_functions_to_scripts <- function(exports_df, pkg_source_path, verbose){
   # Search for scripts functions are defined in
   funcs_df <- get_toplevel_assignments(pkg_source_path)
 
+  if(nrow(funcs_df) == 0){
+    # This shouldn't be triggered, and either indicates a bug in `get_toplevel_assignments`,
+    # or an unexpected package setup that we may want to support.
+    warning("No top level assignments were found in the R/ directory. Make sure this was expected.")
+  }
+
```

Comment:
Given the comments, Im wondering if this should be a `dev_warning()` or something (message about contacting a developer). Either way we may want to discuss this handling

Reply:
At this point, we know there's an `R/` directory, so I agree that not finding any top-level assignments should be pretty rare.

I guess with "contacting a developer" you're talking about the package scorer (likely the package developer) opening an issue here? It's fine by me if you want to link to this issue tracker in the warning (basically what `bbr:::dev_error` and `rlang::abort(..., .internal = TRUE)` do underneath).

Perhaps there are some packages that define every top-level thing with a weird meta-programming approach that we won't want to support, but that should be uncommon enough that directing people won't lead to us having to say "sorry, that's an edge case/approach we don't support" too many times.